cf/data/data.py

```diff
@@ -12463,49 +12446,143 @@ def full(
                 The fill value.
 
             dtype: data-type
-                The data-type of the new array. By default the data-type
-                is ``float``.
+                The data-type of the new array. By default the
+                data-type is ``float``.
```

Comment:
I think it is important to note `optional` on the`dtype` kwarg (even if the 'By default' statement implies this), mainly because our behaviour differs from the `numpy` behaviour here with respect to a given `fill_value`, e.g:

```python
>>> d_cf = cf.Data.full((2, 3), fill_value=10)
>>> d_cf
<CF Data(2, 3): [[10.0, ..., 10.0]]>
>>> d_np = np.full((2, 3), fill_value=10)
>>> d_np
array([[10, 10, 10],
       [10, 10, 10]])
```

That should be sufficient though, as you've included a good equivalent example later in the docstring.

Reply:
Interesting. Couldn't we also follow numpy? and do:
```
    dtype : data-type, optional
        The desired data-type for the array  The default, None, means
         ``np.array(fill_value).dtype``.
```

Reply:
OK - let's go with the numpy behaviour: https://github.com/NCAS-CMS/cf-python/pull/315/commits/8ab7733fdb5e6937ade7f2e770109d2d9678bde6